picard/plugin.py

```diff
@@ -275,6 +275,9 @@ def load_plugindir(self, plugindir):
             name = is_zip(path)
             if not name:
                 name = _plugin_name_from_path(path)
+                # check for symlink
+                if os.path.islink(path):
```

Comment:
Shouldn't the real path passed to `_plugin_name_from_path()` ?

```python
if os.path.islink(path):
    path = os.path.realpath(path)

name = _plugin_name_from_path(path)
```

If it is correct, `islink()` check and following should be moved to `_plugin_name_from_path()`.
And, imho, the comment is not needed, `islink()` is pretty obvious.

Reply:
@zas yes its better to pass original path to the ```_plugin_name_from_path``` as  currently if it is a link then i don't need to call it.

Reply:
@zas one issue arrises while doing this.  while loading the plugin by calling 
```def load_plugin(self, name, plugindir):```
it will use the value of plugindir while it's not correct.

Reply:
One thing we can do is make a tupe in ```names``` list indicating names and directory it is present in. Any thoughts?

Reply:
I had a look at the code, and it looks like it wasn't meant to support symlinks at all ;)

All those name + plugindir splits & join and manipulation are awful.
I'm not sure it is worth trying to support symlinks with current code :(

Reply:
@zas upto you. I will continue to PICARD-1194.